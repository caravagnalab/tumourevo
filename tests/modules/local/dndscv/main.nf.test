nextflow_process {
  name 'Test process DNDSCV'
  script "modules/local/dndscv/main.nf"
  process 'DNDSCV'
  // temporary override for testing
  config 'tests/modules/local/dndscv/test.config'
  
  test("Should run without failures") {
    when {
      process {
        """
        meta = [
            id : "idProva",
            tumour_sample: "Set6_42",
            cancer_type: "PANCANCER" 
        ]
        snv_rds = file('/Users/davide.rambaldi/Projects/nextflow_dev/tumourevo_datasets/VCF.rds')
        driver_list = file('/Users/davide.rambaldi/Documents/Projects/nextflow_dev/tumourevo_datasets/driver_list.txt')
        reference = file('/Users/davide.rambaldi/Documents/Projects/nextflow_dev/tumourevo_datasets/RefCDS_human_GRCh37_dNdScv.0.1.0.rda')

        input[0] = tuple(meta, snv_rds, driver_list, reference)
        """            
      }
    }

    then {
      assert process.success
    }
  }
}
