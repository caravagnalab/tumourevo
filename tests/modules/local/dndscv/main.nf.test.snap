{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_dnds.rds:md5,d4e26a720ac112e6e31140b181cb2c64"
                    ]
                ],
                "dnds_rds": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_dnds.rds:md5,d4e26a720ac112e6e31140b181cb2c64"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-07T09:52:44.220117"
    }
}